### ISSUE TITLE
Notifications should be able to break through closed RECV.

### ISSUE THREAD

**Issue by cbiffle** [oxidecomputer/hubris#1761]
Currently, the reference manual says this about closed-RECV operations:

> During a closed receive, a task will receive messages only from the chosen task. The task will not receive notifications unless the chosen sender ID is the kernel’s task ID, 0xFFFF. (This behavior is a little odd because it predates notification masks, and may change.)

As prophesied, I would like to change this.

Currently we are in the odd situation where entering a closed-RECV with a non-zero notification mask just ignores the notification mask -- it's treated as equivalent to a notification mask of zero. We got here because of some implementation details in the _very early_ kernel ABI (like "first month" early).

This seems awkward. It would probably be better if it were orthogonal:
- Open RECV accepts messages from any task, Closed RECV accepts messages from only one task. (The kernel's virtual task ID would still be valid here, to filter out messages from _all_ user tasks.)
- Notification mask non-zero accepts (some subset of) notifications, zero does not.

Moreover, this may allow some useful behaviors that aren't currently possible. As I noted in #1654 I've been taking a hard look at our closed-RECV operation and thinking about how it interacts with priority. Currently, once a server enters a closed-RECV, it has no way of breaking back out -- timers won't work, interrupts won't work, etc. If we changed closed-RECV to honor the notification mask, it would let servers arrange to break out of a closed-RECV, which would allow them to do all sorts of useful things:
- Process periodic work despite waiting on a client
- Handle interrupts
- Impose a timeout and REPLY-FAULT a client that is taking too long

(I don't think imposing a timeout like that should happen very often, since timeouts are how we make flaky software, but it would be nice to have the ability!)
